great_expectations/rule_based_profiler/helpers/util.py

```diff
@@ -461,6 +463,71 @@ def init_expectation_configuration_builder(
     return expectation_configuration_builder
 
 
+def resolve_evaluation_dependencies(
+    parameter_builder: "ParameterBuilder",  # noqa: F821
+    parameter_container: ParameterContainer,
+    domain: Domain,
+    variables: Optional[ParameterContainer] = None,
+    parameters: Optional[Dict[str, ParameterContainer]] = None,
+) -> None:
+    """
+    This method computes ("resolves") pre-requisite ("evaluation") dependencies (i.e., results of executing other
+    "ParameterBuilder" objects), whose output(s) are needed by specified "ParameterBuilder" object to fulfill its goals.
+    """
+
+    # Step 1: Check if any "evaluation_parameter_builders" are configured for specified "ParameterBuilder" object.
```

Comment:
Loving the comments (super helpful in reading the code) but also wondering if this could be refactored into a few smaller methods. Similar thougts with `get_metric()` actually.

Reply:
@Shinnnyshinshin I would not want to disturb `get_metric()` now.  In terms of this one, perhaps the comments make it parseable in small enough parts.  It is pretty much a single thought in that method.